lib/routing/QubitMetadataRoute.class.php

```diff
@@ -77,17 +77,8 @@ public function matchesUrl($url, $context = [])
             $criteria->add(QubitSlug::SLUG, $parameters['slug']);
             $criteria->addJoin(QubitSlug::OBJECT_ID, QubitObject::ID);
 
-            try {
-                if (null === $this->resource = QubitObject::get($criteria)->__get(0)) {
-                    return false;
-                }
-            }
-            // If for any reason the database can't be accessed, trigger the installer
-            catch (PropelException $e) {
-                $parameters['module'] = 'sfInstallPlugin';
-                $parameters['action'] = 'index';
-
-                return $parameters;
+            if (null === $this->resource = QubitObject::get($criteria)->__get(0)) {
+                return false;
```

Comment:
What happens in this case? 500 error if arInstall has not completed?

Reply:
I have not tested it, but that was my assumption. I'll give it a try.

Reply:
Good thing you pointed this out as it's quite confusing. Based on the URL you hit, when you go to the site without running the installer you get different errors ...

- Using a slug you get our normal 500 error page and the following error in the logs:

```
[11-Jun-2021 17:27:49] WARNING: [pool atom] child 16 said into stderr: "NOTICE: PHP message: No connection information in your runtime configuration file for datasource [propel]"
```

- Without slug or using a module/action route, you get a default 500 error page and a bit more confusing log:

```
[11-Jun-2021 17:21:02] WARNING: [pool atom] child 16 said into stderr: "NOTICE: PHP message: Empty module and/or action after parsing the URL "/" (/)."
[11-Jun-2021 17:21:02] WARNING: [pool atom] child 16 said into stderr: "NOTICE: PHP message: PHP Fatal error:  Uncaught sfError404Exception: Action "default/error404" does not exist. in /atom/src/cache/qubit/prod/config/config_core_compile.yml.php:660"
[11-Jun-2021 17:21:02] WARNING: [pool atom] child 16 said into stderr: "Stack trace:"
[11-Jun-2021 17:21:02] WARNING: [pool atom] child 16 said into stderr: "#0 /atom/src/vendor/symfony/lib/exception/sfError404Exception.class.php(49): sfController->forward('default', 'error404')"
[11-Jun-2021 17:21:02] WARNING: [pool atom] child 16 said into stderr: "#1 /atom/src/cache/qubit/prod/config/config_core_compile.yml.php(2396): sfError404Exception->printStackTrace()"
[11-Jun-2021 17:21:02] WARNING: [pool atom] child 16 said into stderr: "#2 /atom/src/vendor/symfony/lib/util/sfContext.class.php(170): sfFrontWebController->dispatch()"
[11-Jun-2021 17:21:02] WARNING: [pool atom] child 16 said into stderr: "#3 /atom/src/index.php(6): sfContext->dispatch()"
[11-Jun-2021 17:21:02] WARNING: [pool atom] child 16 said into stderr: "#4 {main}"
172.18.0.4 -  11/Jun/2021:17:21:02 +0000 "GET /index.php" 500
[11-Jun-2021 17:21:02] WARNING: [pool atom] child 16 said into stderr: "  thrown in /atom/src/cache/qubit/prod/config/config_core_compile.yml.php on line 660"
[11-Jun-2021 17:21:49] WARNING: [pool atom] child 15 said into stderr: "NOTICE: PHP message: No connection information in your runtime configuration file for datasource [propel]"


[11-Jun-2021 17:28:33] WARNING: [pool atom] child 15 said into stderr: "NOTICE: PHP message: Empty module and/or action after parsing the URL "/actor/" (/)."
[11-Jun-2021 17:28:33] WARNING: [pool atom] child 15 said into stderr: "NOTICE: PHP message: PHP Fatal error:  Uncaught sfError404Exception: Action "default/error404" does not exist. in /atom/src/cache/qubit/prod/config/config_core_compile.yml.php:660"
172.18.0.4 -  11/Jun/2021:17:28:33 +0000 "GET /index.php" 500
[11-Jun-2021 17:28:33] WARNING: [pool atom] child 15 said into stderr: "Stack trace:"
[11-Jun-2021 17:28:33] WARNING: [pool atom] child 15 said into stderr: "#0 /atom/src/vendor/symfony/lib/exception/sfError404Exception.class.php(49): sfController->forward('default', 'error404')"
[11-Jun-2021 17:28:33] WARNING: [pool atom] child 15 said into stderr: "#1 /atom/src/cache/qubit/prod/config/config_core_compile.yml.php(2396): sfError404Exception->printStackTrace()"
[11-Jun-2021 17:28:33] WARNING: [pool atom] child 15 said into stderr: "#2 /atom/src/vendor/symfony/lib/util/sfContext.class.php(170): sfFrontWebController->dispatch()"
[11-Jun-2021 17:28:33] WARNING: [pool atom] child 15 said into stderr: "#3 /atom/src/index.php(6): sfContext->dispatch()"
[11-Jun-2021 17:28:33] WARNING: [pool atom] child 15 said into stderr: "#4 {main}"
[11-Jun-2021 17:28:33] WARNING: [pool atom] child 15 said into stderr: "  thrown in /atom/src/cache/qubit/prod/config/config_core_compile.yml.php on line 660"
```

I'm trying to normalize this,  but it's not as easy as it looks ... thoughts?

Reply:
Previously we redirected to sfInstall when things weren't set up. Could we detect if the installer has not been run and  redirect to a static "Go run the AtoM CLI installer" page for all routes?

Reply:
Ok, still a normal error page but now the error will be the same for all routes. In any case, lets hope they don't forget to run the installer.

Reply:
Addressed in fa2a1dd and 2df334e.